osv/impact.py

```diff
@@ -59,24 +60,25 @@ class RangeCollector:
   def __init__(self):
     self.grouped_ranges = {}
 
-  def add(self, introduced_in, fixed_in):
+  def add(self, introduced_in, fixed_in, affected_in):
     """Add a new commit range."""
     if introduced_in in self.grouped_ranges:
-      if fixed_in is None:
+      if fixed_in is None and affected_in is None:
         # New range doesn't add anything new.
         return
 
       # Remove in-place as we need to preserve insertion order.
       ranges = self.grouped_ranges[introduced_in]
-      ranges.append((introduced_in, fixed_in))
+      ranges.append((introduced_in, fixed_in, affected_in))
       for value in ranges.copy():
         if value[1] is None:
```

Comment:
I don't fully understand what this function is doing, but I think this should now be
```python
if value[1] is None and value[2] is None:
```

Reply:
I'm finding why a method named `add` is _removing_ anything to be... warranting more comments explaining what's going on here. I agreee with you on the addition, on further inspection. @oliverchang can you provide some more context so I can add more comments to the code while I'm here?

Reply:
Addressed in f4072fe and renamed some variables for better readability

Reply:
This is to address a limitation with tuples being immutable. We remove them rather than setting them in place.

Reply:
> I don't fully understand what this function is doing, but I think this should now be
> 
> ```python
> if value[1] is None and value[2] is None:
> ```

I *think* @oliverchang did what needed to be done in 780ad77 so this has been addressed appropriately.